server/src-lib/Hasura/GraphQL/Transport/WebSocket/Server.hs

```diff
@@ -176,12 +184,16 @@ createServerApp (WSServer logger@(L.Logger writeLog) connMap) wsHandlers pending
         WS.sendTextData conn msg
         writeLog $ WSLog wsId $ EMessageSent $ TBS.fromLBS msg
 
+      closeRef <- A.async $ do
+        _ <- STM.atomically $ STM.readTChan shutdownChan
```

Comment:
Please don’t use `_ <-` if the result is `()`, anyway (and if it isn’t, prefer `void`).